### DISCUSSION TITLE
UncontrolledProviderScope Error

### DISCUSSION THREAD

**Discussion by fadodr** [rrousselGit/riverpod#1571] [Q&A]
why do i keep getting this error while pumping my widget in widget testing

await tester.pumpWidget(ProviderScope(
      child: MaterialApp(
        home: LayoutBuilder(
          builder: (context, constraints) {
            return OnboardingImage(constraints);
          }
        ),
      ),

this is the onboardingimage
for (int i = 0; i < 3; i++)
            SizedBox.expand(
              child: Consumer(
                builder: (context, ref, _) {
                  final onboardingController = ref.watch(onboardingProvider);
                  return AnimatedOpacity(
                    opacity: onboardingController.currentPage == i ? 1 : 0,
                    duration: const Duration(milliseconds: 700),
                    child: Image.asset(
                      onboardingController.onboardingToDisplay.image,
                      fit: BoxFit.cover,
                    ),
                  );
                },
              ),
            ),

this is the error 
The following LateError was thrown building Consumer(dirty, dependencies:
[UncontrolledProviderScope], state: _ConsumerState#0b731):
LateInitializationError: Field 'onboardingToDisplay' has not been initialized.
@rrousselGit @TimWhiting

**Comment by TimWhiting** [comment]
Not enough information to answer. But it is not a problem with riverpod. A field in your onboarding controller isn't being initialized but is marked late.
